src/test/java/org/prebid/server/bidder/screencore/ScreencoreBidderTest.java

```diff
@@ -0,0 +1,203 @@
+                });
+    }
+
+    @Test
+    public void makeHttpRequestsShouldMakeCorrectRequest() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                identity(),
+                impBuilder -> impBuilder.id("imp_id2").ext(givenImpExt()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        final BidRequest expectedBidRequest = givenBidRequest(
+                impBuilder -> impBuilder.ext(null),
+                impBuilder -> impBuilder.id("imp_id2").ext(givenImpExt()));
+        final MultiMap expectedHeaders = headers()
+                .set(CONTENT_TYPE_HEADER, APPLICATION_JSON_CONTENT_TYPE)
+                .set(ACCEPT_HEADER, APPLICATION_JSON_VALUE)
+                .set(X_OPENRTB_VERSION_HEADER, "2.5")
+                .set(USER_AGENT_HEADER, "ua")
+                .set(X_FORWARDED_FOR_HEADER, List.of("ipv6", "ip"));
+        final Result<List<HttpRequest<BidRequest>>> expectedResult = Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri("http://h1.screencore.io/?kp=accountId&kn=placementId")
+                .headers(expectedHeaders)
+                .impIds(Set.of("imp_id", "imp_id2"))
+                .body(jacksonMapper.encodeToBytes(expectedBidRequest))
+                .payload(expectedBidRequest)
+                .build()
+        );
+        assertThat(result.getValue()).usingRecursiveComparison().isEqualTo(expectedResult.getValue());
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorWhenResponseCanNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> actual = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Bad Server Response"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorWhenResponseDoesNotHaveSeatBid() throws JsonProcessingException {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, givenBidResponse());
+
+        // when
+        final Result<List<BidderBid>> actual = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Empty SeatBid array"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnBidsSuccessfully() throws JsonProcessingException {
+        // given
+        final Bid bannerBid = Bid.builder().impid("1").mtype(1).build();
+        final Bid videoBid = Bid.builder().impid("2").mtype(2).build();
+        final Bid nativeBid = Bid.builder().impid("3").mtype(4).build();
```

Comment:
Separate to tests with only one mediatype check

Reply:
I believe the current test makes sense, it's possible to receive all three bids in the same response, isn't it?

perhaps what you suggest is adding more test cases that covers single bid cases... I can add them, if you think the current test is not enough